In [1]:
import pandas as pd
import numpy as np


In [2]:
import scipy.stats as stats

In [3]:
from scipy.stats import ttest_1samp,ttest_ind,ttest_rel


## Excercise 1

In [11]:
data_1 = pd.read_csv('CUtlets.csv')

In [12]:
ttest_ind(data_1["Unit A"].dropna(), data_1["Unit B"].dropna(), alternative ='two-sided')

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

+ p > 0.05 , so we fail to reject the null hypothesis which means there is no significant difference in the diameter of cutlets from Unit A & B

## Excercise 2

In [5]:
data_2 = pd.read_csv('LabTAT.csv')

In [14]:
stats.f_oneway(data_2.iloc[:,0],data_2.iloc[:,1],data_2.iloc[:,2],data_2.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

+ p < 0.05 , hence there is atleast one difference in average TAT among the four laboratories

## Excercise 3

In [19]:
data_3 = [[50,142,131,70],[550,351,480,350]]
data_3

[[50, 142, 131, 70], [550, 351, 480, 350]]

In [20]:
stats,p,dof,expected = stats.chi2_contingency(data_3)
print(dof)
p

3


2.682172557281901e-17

+ p<0.05, Hence not all proportions are equal

## Excercise 4

In [4]:
data_4 = pd.read_csv('Costomer+OrderForm.csv')

In [5]:
data_4.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [7]:
Phillippines = data_4['Phillippines'].value_counts()
Phillippines

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [8]:
Indonesia = data_4['Indonesia'].value_counts()
Indonesia

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [9]:
Malta = data_4['Malta'].value_counts()
Malta

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [10]:
India = data_4['India'].value_counts()
India

Error Free    280
Defective      20
Name: India, dtype: int64

In [17]:
stats_1 =stats.chi2_contingency([[271,267,269,280],[29,33,31,20]])
stats_1

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

+ p = 0.277 > 0.05, Hence we accept the null hypothesis which is defective % across the center is same